src/runtime.js

```diff
@@ -359,6 +370,27 @@ let tornado = {
   },
 
   /**
+   * Build a pending body within a div with class "pending" (we have to wrap in a div so we can
+   * easily replace the entire thing when the promise resolves). If no pending body exists, then
+   * return false.
+   * @param {Node} placeholderNode The node where the pending body will be inserted
+   * @param {[Function]} body The pending body function, if it exists
+   * @param {Object} context The current Tornado context, to be used in building the pending body
+   * @return {HTMLElement|False} Return the containing div, or false
+   */
+  insertPendingBody(placeholderNode, body, context) {
+    if (body) {
+      let div = document.createElement('div');
+      div.setAttribute('class', 'pending');
```

Comment:
This is necessary, unless we want to add some other kind of ugly decoration. Here is the problem that the `div.pending` wrapper solves:

Fragment before inserting `pending` body:
```
My content will go here: [single empty string text node, acting as an insertion point]
```
Fragment after inserting pending body:
```
My content will go here: While you wait, here's a <b>really</b> funny joke. Why did the rubber band ...
```
Now, where we had a single, empty-string text node we have 3 sibling nodes: text, <b>, and text. When my content arrives, we need to insert it, but where? Our insertion point is gone, replaced by 3 text nodes.

In theory, we could count how many top level nodes are in a given pending body, and then replace that many nodes when the real content arrives, but that is a lot of overhead, and quite messy.

So, instead, we wrap the pending body in a div (with a convenient class name), and we maintain the single insertion point, and we document the fact that a pending body gets a wrapper. I think this div wrapper can be viewed as a feature rather than a nuisance. An empty pending body will give you div in the right place, where you can put a loading spinner, or whatever you want using CSS.